src/rgw/rgw_aws.cc

```diff
@@ -0,0 +1,259 @@
+      invokeRequest.SetLogType(Aws::Lambda::Model::LogType::Tail);
+      invokeRequest.SetFunctionName(arn.GetResource());
+      std::shared_ptr<Aws::IOStream> body = Aws::MakeShared<Aws::StringStream>("Simple Allocation Tag");
+      *body << payload;
+      invokeRequest.SetBody(body);
+      invokeRequest.SetContentType("application/json");
+      auto response = client->Invoke(invokeRequest);
+      if (!response.IsSuccess()) {
+        ldout(cct, 1) << response.GetError().GetMessage() << dendl;
+      } else {
+        Aws::Lambda::Model::InvokeResult &result = response.GetResult();
+        Aws::IOStream &pl = result.GetPayload();
+        Aws::String returnedData;
+        std::getline(pl, returnedData);
+        ldout(cct, 1) << "Success " << returnedData << "\n" << dendl;
+      }
+    }
+
+//    Works on messages in the queque
+    void run() {
+      while (!stopped) {
+        const auto count = messageQueue.consume_all(
+                std::bind(&Manager::publish_internal, this, std::placeholders::_1));
+        if (count == 0) {
+          std::this_thread::sleep_for(idle_time);
+        }
+      }
+    }
+
+    // used in the dtor for message cleanup
+    static void delete_message(const message_t *message) {
+      delete message;
+    }
+
+  public:
+    Manager(size_t _max_clients, size_t max_queue_size,unsigned idle_time_ms, CephContext *_cct) : messageQueue(max_queue_size),
+                                                                                                   clientList(_max_clients),
+                                                                                                   runner(&Manager::run, this),
+                                                                                                   stopped(false), cct(_cct),
+                                                                                                   idle_time(std::chrono::milliseconds(idle_time_ms)){}
+    // dtor wait for thread to stop
+    // then connection are cleaned-up
+    ~Manager() {
+      stopped = true;
+      runner.join();
+      messageQueue.consume_all(delete_message);
+    }
+
+
+    AwsClient connect(const std::string &accessKey,
+                      const std::string &accessSecret,
+                      const std::string &arn,
+                      const std::string &caPath,
+                      bool useSSL) {
+      if (stopped) {
+        return AwsClient();
+      }
+      const client_id_t id(accessKey, accessSecret, arn, useSSL);
+      std::lock_guard lock(clientsLock);
+      const auto it = clientList.find(id);
+      if (it != clientList.end()) {
+        return it->second;
+      }
+
+      // connection not found, creating a new one
+      Aws::Client::ClientConfiguration configuration;
+      if (useSSL) {
+        configuration.scheme = Aws::Http::Scheme::HTTPS;
+        configuration.verifySSL = true;
```

Comment:
you can take the value of `verifySSL` from the topic parameters. this will help when testing against a mock AWS server